proposals/low-level-struct-improvements.md

```diff
@@ -849,24 +787,129 @@ escapes struct S
+
+In brief, the two proposed compat changes are:
+
+* A `ref` parameter on method returning a `ref struct` would be capturable as a `ref` field
+* A `out` parameter would be considered `ref-safe-to-escape` within the *current method*.
+
+In this design a `ref` field in a constructor can be effectively ref reassigned to any value that is *ref-safe-to-escape* to the *calling method*. Specifically the rules for ref reassignment will be adjusted to account for `ref` fields as follows:
+
+> For a ref reassignment in the form ...
+> 1. `x.e1 = ref e2`: where `x` is *safe-to-escape* to *calling method* then `e2` must be *ref-safe-to-escape* to the *calling method*
+> 2. `e1 = ref e2`: the *ref-safe-to-escape* of `e2` must be at least as wide a scope as the *ref-safe-to-escape* of `e1`.
+
+That means the desired `Span<T>` constructor works without any extra annotation:
 
-Consider for a minute where the design was approached from the other direction: APIs with `ref` fields have the ideal state and burden is placed on developers with incorrect APIs to fix them up when compiling for C# 11. This compat break would take primarily two forms:
+```c#
+Span(ref T value)
+{
+    // `this` is safe-to-escape and `value` is ref-safe-to-escape to calling method hence 
+    // legal by rule 1 above
+    _field = ref value;
+}
+```
+
+The keyword `scoped` will be used to restrict the lifetime of a value. It can be applied to a `ref` or a value and has the impact of restricting the *ref-safe-to-escape* or *safe-to-escape* lifetime, respectively, to the *current method*. For example: 
+
+| Parameter | ref-safe-to-escape | safe-to-escape |
+|---|---|---|
+| `Span<int> s` | *current method* | *calling method* | 
+| `scoped Span<int> s` | *current method* | *current method* | 
+| `ref Span<int> s` | *calling method* | *calling method* | 
+| `scoped ref Span<int> s` | *current method* | *calling method* | 
+| `ref scoped Span<int> s` | *current method* | *current method* | 
+
+The declaration `scoped ref scoped Span<int>` is allowed but is redundant with `ref scoped Span<int>`. The *ref-safe-to-escape* of a value can never exceed the *safe-to-escape* hence once the *safe-to-escape* is restricted via `ref scoped` the *ref-safe-to-escape* is implicitly restricted.
+
+This also means that the `this` parameter of a `struct` can now be completely expressed as `scoped ref T`. Previously it had to be special cased as a `ref T` that had different *ref-safe-to-escape* rules than other `ref`. Now it can be expressed as a general concept vs. being a special cased item.
+
+These annotations can be applied to locals as well as parameters and have the same impact to lifetimes. They cannot be applied to any other location including fields, array elements, etc ... Further while `scoped` can be applied to any `ref` or `in` it can only be applied to values which are `ref struct`. Having `scoped int` adds no value to the rules and will be prevented to avoid confusion.
```

Comment:
```suggestion
These annotations can be applied to locals as well as parameters and have the same impact to lifetimes. They cannot be applied to any other location including fields, array elements, etc ... Further while `scoped` can be applied to any `ref` or `in` it can only be applied to by-values which are `ref struct`. Having `scoped int` adds no value to the rules and will be prevented to avoid confusion.
```

I found the "values" a bit confusing.